File: taskflow-fullstack/frontend/src/services/dashboard-service.ts
import { apiRequest } from '../lib/api-client'
import type { DashboardSummaryResponse, DashboardTaskItem } from '../types/dashboard'

export const dashboardService = {
  async getSummary(): Promise<DashboardSummaryResponse> {
    return apiRequest<DashboardSummaryResponse>('/api/v1/dashboard/summary', {
      method: 'GET',
      authenticated: true,
    })
  },

  async getMyTasks(): Promise<DashboardTaskItem[]> {
    return apiRequest<DashboardTaskItem[]>('/api/v1/dashboard/my-tasks', {
      method: 'GET',
      authenticated: true,
    })
  },

  async getUpcomingDeadlines(): Promise<DashboardTaskItem[]> {
    return apiRequest<DashboardTaskItem[]>('/api/v1/dashboard/upcoming-deadlines', {
      method: 'GET',
      authenticated: true,
    })
  },
}

File: taskflow-fullstack/backend/src/main/java/com/ruhatkaratas/taskflow/dashboard/service/DashboardService.java
package com.ruhatkaratas.taskflow.dashboard.service;

import com.ruhatkaratas.taskflow.dashboard.dto.DashboardSummaryResponse;
import com.ruhatkaratas.taskflow.dashboard.dto.DashboardTaskItem;
import java.util.List;

public interface DashboardService {

    DashboardSummaryResponse getSummary(String currentUserEmail);

    List<DashboardTaskItem> getMyTasks(String currentUserEmail);

    List<DashboardTaskItem> getUpcomingDeadlines(String currentUserEmail);
}

